quick_sort.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import unicode_literals
+from time import time
+from random import shuffle
+
+
+def quick_sort(init_list):
+    return_list = init_list[:]
+    _quick_sort2(return_list, 0, len(return_list) - 1)
+    return return_list
+
+
+def _quick_sort2(init_list, low, high):
+    if low < high:  # if there's more than one item to be sorted
+        p = _partition(init_list, low, high)
+        _quick_sort2(init_list, low, p - 1)
+        _quick_sort2(init_list, p + 1, high)
+
+
+def _get_pivot(init_list, low, high):
+    mid_idx = (high + low) // 2
+    pivot = high
+    if init_list[low] < init_list[mid_idx]:
+        if init_list[mid_idx] < init_list[high]:
+            pivot = mid_idx
+    elif init_list[low] < init_list[high]:
+        pivot = low
+    return pivot
```

Comment:
kind of an awkward way to return the middle of three numeric values.  And does it always work?